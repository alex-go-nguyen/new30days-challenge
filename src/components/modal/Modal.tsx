import React, { PropsWithChildren } from 'react';
import styles from './Modal.module.scss';
import { IoMdClose } from 'react-icons/io';
import Button from '../button/Button';
import classNames from 'classnames';

type ModalType = 'success' | 'error';

export interface ModalProps {
    isOpen?: boolean;
    type?: ModalType;
    onClose?: () => void;
    title: string;
}

export default function Modal({ isOpen, onClose, title, type = 'success', children }: PropsWithChildren<ModalProps>) {
    return (
        <div>
            <div className={classNames({ [styles.modal]: true, [styles.isOpened]: isOpen })}>
                <div className={styles.container}>
                    <div className={classNames({ [styles.heading]: true, [styles[type]]: true })}>
                        <p className={styles.title}>{title}</p>
                        <span className={styles.closeIcon} onClick={onClose}>
                            <IoMdClose />
                        </span>
                    </div>
                    <div className={styles.content}>
                        {children}
                        <div className={styles.closeBtn} onClick={onClose}>
                            <Button variant="primary">Close</Button>
                        </div>
                    </div>
                </div>
            </div>
        </div>
    );
}
