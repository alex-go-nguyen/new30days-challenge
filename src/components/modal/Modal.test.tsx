import React from 'react';
import renderer from 'react-test-renderer';
import Modal from './Modal';

describe('Modal component', () => {
    const component = renderer.create(<Modal title="Success">Hello success</Modal>);
    let tree = component.toJSON();
    test('should render correctly', () => {
        expect(tree).toMatchSnapshot();
    });
});
