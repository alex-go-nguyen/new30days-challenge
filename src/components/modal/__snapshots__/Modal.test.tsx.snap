// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal component should render correctly 1`] = `
<div>
  <div
    className="modal"
  >
    <div
      className="container"
    >
      <div
        className="heading success"
      >
        <p
          className="title"
        >
          Success
        </p>
        <span
          className="closeIcon"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              {
                "color": undefined,
              }
            }
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M405 136.798L375.202 107 256 226.202 136.798 107 107 136.798 226.202 256 107 375.202 136.798 405 256 285.798 375.202 405 405 375.202 285.798 256z"
            />
          </svg>
        </span>
      </div>
      <div
        className="content"
      >
        Hello success
        <div
          className="closeBtn"
        >
          <button
            className="primary button md success"
            type="button"
          >
            Close
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
